Library/ViewModels/SelectCurrencyViewModel.swift

```diff
@@ -33,6 +39,51 @@ SelectCurrencyViewModelOutputs {
       initialChosenCurrency,
       self.didSelectCurrencySignal
     )
+
+    let updateCurrencyEvent = self.selectedCurrencyProperty.signal.skipNil()
+      .takeWhen(self.saveButtonTappedSignal.ignoreValues())
+      .switchMap { input in
+        AppEnvironment.current.apiService
+          .changeCurrency(input: ChangeCurrencyInput(chosenCurrency: input.rawValue))
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .map { _ in input }
+          .materialize()
+    }
+
+    self.updateCurrencyDidSucceed = updateCurrencyEvent.values().ignoreValues()
+
+    self.updateCurrencyDidFailWithError = updateCurrencyEvent.errors().map { $0.localizedDescription }
+
+    self.activityIndicatorShouldShow = Signal.merge(
+      self.saveButtonTappedSignal.mapConst(true),
+      updateCurrencyEvent.filter { $0.isTerminating }.mapConst(false)
+    )
+
+    let updatedCurrency = updateCurrencyEvent.values()
+
+    updatedCurrency
+      .observeValues { AppEnvironment.current.koala.trackChangedCurrency($0) }
```

Comment:
Do we also need a koala event for the view loading?

Reply:
Possibly - will find out from @nnekab what the event would be for that - there wasn't one before because it was a picker.